### PR TITLE
[FIX][TST] Adds curl installation where needed

### LINKED ISSUE #388: Patch coverage reporting
### Summary
Coverage reports are no longer uploading from CircleCI to CodeCov.

### Additional Detail
Specifically, the upload step fails with the following:
```
#!/bin/bash -eo pipefail
curl -s https://codecov.io/bash | bash -s -- \
  -f "/tmp/src/coverage/coverage.xml" \
  -t "${CODECOV_TOKEN}" \
  -n "${CIRCLE_BUILD_NUM}" \
  -y ".codecov.yml" \
  -F "" \
  -Z || echo 'Codecov upload failed'

/bin/bash: curl: command not found
Codecov upload failed
```

### Next Steps

I'm planning to open a PR to patch this. @tsalo since you most recently modified the CircleCI config, I might ask for your review !

### PR DESCRIPTION
**Pull request by jbteves**
Closes #388 .

Changes proposed in this pull request:

- Installs curl where needed
-

### CODE REVIEW

**Comment by jbteves** [comment]
@tsalo I don't know why it says I cancelled your review but I do in fact want it.

File: .circleci/config.yml [hunk 1]
```diff
@@ -277,6 +281,7 @@ jobs:
       - run:
           name: Merge coverage files
           command: |
+            apt-get install -y curl
```
Comment (tsalo) [inline .circleci/config.yml]: Is this one necessary?

Comment (jbteves) [reply .circleci/config.yml]: Unbelievably, yes. Codecov uses curl to upload the report, that’s why it kept failing.

Comment (tsalo) [reply .circleci/config.yml]: Weird, but good to know.

**Pull request merged**